gatsby-node.js

```diff
@@ -1,14 +1,72 @@
 /* eslint-disable import/order */
+const webpack = require('webpack')
+const { merge } = require('webpack-merge')
 const language_config = require(`./i18n-config.js`)
 const language_config_en = require(`./i18n-config-en.js`)
 const path = require('path')
 const { copyLibFiles } = require('@builder.io/partytown/utils')
-const webpack = require('webpack')
 const StylelintPlugin = require('stylelint-webpack-plugin')
 const TerserPlugin = require('terser-webpack-plugin')
 
 const translations_cache = {}
 
+const commonConfig = ({ stage, actions, loaders, getConfig }, { ...options }) => {
+    const config = getConfig();
+    const isProduction = config.mode === 'production';
+
+    const baseConfig = {
+        devtool: isProduction ? false : 'inline-source-map',
+        mode: isProduction ? 'production' : 'development',
+        optimization: {
+            minimize: isProduction,
+            minimizer: [new TerserPlugin()],
+            splitChunks: {
+                chunks: 'all',
+                cacheGroups: {
+                    default: false,
+                    vendors: false,
+                    // Merge all js, ts, and tsx files into one bundle
+                    all: {
+                        test: /\.(js|ts|tsx)$/,
+                        name: 'deriv',
+                        chunks: 'all',
+                    },
+                },
+            },
+            mangleExports: 'size',
+            mangleWasmImports: true,
+            mergeDuplicateChunks: true,
+            removeAvailableModules: true,
+            removeEmptyChunks: true,
+            innerGraph: true,
+            chunkIds: 'size',
+            moduleIds: 'size',
+            sideEffects: true,
+            concatenateModules: true,
+            providedExports: true,
+            usedExports: true,
+        },
+        plugins: [
+            new StylelintPlugin({ ...style_lint_options, ...options }),
+            new webpack.optimize.LimitChunkCountPlugin({maxChunks: 1}),
+        ],
+        resolve: {
+            modules: [path.resolve(__dirname, 'src'), 'node_modules'],
+        },
+    };
+
+    const htmlConfig = (stage === 'build-html' || stage === 'develop-html') ? {
+        module: {
+            rules: [{
+                test: /analytics/,
+                use: loaders.null(),
+            }],
+        },
+    } : {};
+
+    return merge(baseConfig, htmlConfig);
+}
+
```

Comment:
Usually, the function declaration goes bellow the call. So you as a developer read important things first and details later.